### DISCUSSION TITLE
How to open VS Code with a Tauri App?

### DISCUSSION THREAD

**Discussion by pt-hieu** [tauri-apps/tauri#5053] [Q&A] [answered]
Hi all,

I'm developing a new Tauri app that lets me open a VS Code codespace with a given path. I'm doing like:

```typescript
  if (isWeb()) return

  const path = repo?.value?.repo.path
  if (!path) {
    toast.add('Path has not been configured', 'Error', undefined, 2)
    return
  }

  const cmd = new Command('code', path)
  // const cmd = new Command('code', '.', { cwd: path }) 
  cmd.on('error', () => {
    toast.add('Opening repo in Vs Code failed!', 'Error', OPEN_VSCODE_TOAST_ID, 2)
  })

  await cmd.spawn().catch((e) => {
    alert(e)
  })
```

My tauri config for that is
```json
{
   "name": "code",
   "cmd": "code",
   "args": true
}
```

What I have received is `program not found`. I also tried `echo %PATH%` and still had the same output. So I have no idea why this is happening or if it's relating to the `sidecar` feature of Tauri. How am I gonna deal with it?

Thanks

**Comment by FabianLars** [comment, marked as answer]
The problem with vscode in particular is that what's added to the PATH var is not the exe itself, but a cmd script. This means that, at least on windows, you have to explicitly call that file by changing the scope config's cmd to `"cmd": "code.cmd"`.
On Linux/macos it's a bit tricky since the bundled app packages don't even have inherit the PATH from shell dotfile (you can regain access to it via this crate https://github.com/tauri-apps/fix-path-env-rs). But on these platforms you probably would invoke the equivalent to `sh -c "code path"` to start vscode (which should work on windows too btw)

**Reply by pt-hieu** [reply]
got it, thank you
